[receivers/mongodbatlasrecevier] fix first scrape lookback time

mstumpfx opened the pull request:
**Description:** The first `start` lookback was being set to the 0 value of time.Time, and ultimately being passed to the mongodb atlas api with a value of `0001-01-01T00:00:00Z`, which is `bad`. Results in an attempt to query a large amount of data.

Commit:
[receivers/mongodbatlasrecevier] fix first scrape lookback time

anuraaga reviewed (commented):
Can you add a unit test to verify the behavior?

Commit:
refactor timeconstraint and add receiver_test

Commit:
update changelog

mstumpfx commented:
> Can you add a unit test to verify the behavior?
@anuraaga I added a attempted to split out the functionality to make it easier to test, and added `receiver_test.go` with a unit test for it.

dmitryax commented on receiver/mongodbatlasreceiver/receiver_test.go:
```diff
@@ -0,0 +1,86 @@
+				require.NotNil(t, tc)
+				require.Equal(t, tc.start, now.Add(cfg.CollectionInterval*-1).UTC().Format(time.RFC3339))
+				// set lookback for next test after this one is done
+				recv.lastRun = now
```
It doesn't seem right to make one test case dependent on another. I would recommend extracting subtests into a plain test case or making them independent.

mstumpfx replied:
I was thinking the same thing when I wrote it tbh, I'll split it out now.

mstumpfx replied:
@dmitryax updated to make them independent.

dmitryax reviewed (commented):
Just one nit. otherwise LGTM

jpkrohling committed:
Merge branch 'main' into mdb-atlas
